### PR TITLE
Bullet behaviour refactor

### LINKED ISSUE #33: Refactor HitBehaviour, BulletBehaviour, and BulletMono
We need to move code out of BulletMono into BulletBehaviour, and update IHitBehaviour with relevant abstract methods.

### PR DESCRIPTION
**Pull request by Sirkular**
* Renamed IHitBehaviour to AbstractAttackBehaviour and other properties for clarity
* AbstractAttackBehaviour now contains damage ~~and onHit~~ as well as a constructor to set owner and damage
* BulletAttackBehaviour now contains most attack behaviour, with BulletMono only acting as a vehicle to determine hits

Closes #33

### CODE REVIEW

**Review by zhuandy42 — APPROVED** [review]
lgtm

good call on renaming IHitBehaviour

**Comment by MaXeraph** [comment]
Looks like we aren't using the modifiers yet right

File: Assets/Scripts/Weapons/BehaviourTypes/BulletMono.cs [hunk 1]
```diff
@@ -2,50 +2,31 @@
...
-        Stats statsComponent = c.gameObject.GetComponent<Stats>();
-        if (statsComponent){
-            if (statsComponent.owner != _owner){
-                statsComponent.takeDamage(_damage);
-                Destroy(gameObject); // bullet
-            }
-        } else {
+    void OnTriggerEnter(Collider c)
+    {
+        if (!_attackBehaviour.onHit(c.gameObject))
```
Comment (MaXeraph) [inline Assets/Scripts/Weapons/BehaviourTypes/BulletMono.cs:27]: I feel like we should rename `onHit` to something else, it returning `True` or `False` isn't really intuitive to me. Unless I'm missing something.
Or maybe flip the boolean so it means if it `hit` something it should destroy itself

Comment (Sirkular) [reply Assets/Scripts/Weapons/BehaviourTypes/BulletMono.cs:27]: I thought about this as well, and I agree it's unintuitive to have onHit return a boolean for whether or not the attack should continue. It's not particularly applicable to all attacks. I'm going to change this shortly.

**Comment by Sirkular** [comment]
> Looks like we aren't using the modifiers yet right

Yup. Should be implemented when we work on stat mods.

**Comment by Sirkular** [comment]
onHit was an odd method to have in AbstractAttackBehaviour, while all attacks should have an effect on hit, there was no reason to have it be an abstract method that _must_ be implemented. Each attack will implement this, but having the signature there only limited implementation.

**Review by MaXeraph — APPROVED** [review]

**Pull request merged**